typespec-extension/src/code-model-builder.ts

```diff
@@ -1104,68 +1059,70 @@ export class CodeModelBuilder {
       // format if array
       let style = undefined;
       let explode = undefined;
-      if (param.param.type.kind === "Model" && isArrayModelType(this.program, param.param.type)) {
-        if (param.type === "query") {
-          // eslint-disable-next-line deprecation/deprecation
-          const queryParamFormat = param?.format;
-          if (queryParamFormat) {
-            switch (queryParamFormat) {
-              case "csv":
-                style = SerializationStyle.Simple;
-                break;
-
-              case "ssv":
-                style = SerializationStyle.SpaceDelimited;
-                break;
-
-              case "tsv":
-                style = SerializationStyle.TabDelimited;
-                break;
-
-              case "pipes":
-                style = SerializationStyle.PipeDelimited;
-                break;
-
-              case "multi":
-                style = SerializationStyle.Form;
-                explode = true;
-                break;
-
-              default:
-                this.logWarning(`Unrecognized query parameter format: '${queryParamFormat}'.`);
-                break;
-            }
+      if (sdkType.kind === "array") {
+        if (param.kind === "query") {
+          const format = param.collectionFormat;
+          switch (format) {
+            case "csv":
+            case "simple":
+              style = SerializationStyle.Simple;
+              break;
+
+            case "ssv":
+              style = SerializationStyle.SpaceDelimited;
+              break;
+
+            case "tsv":
+              style = SerializationStyle.TabDelimited;
+              break;
+
+            case "pipes":
+              style = SerializationStyle.PipeDelimited;
+              break;
+
+            case "multi":
+            case "form":
+              style = SerializationStyle.Form;
+              explode = true;
+              break;
+
+            default:
+              if (format) {
```

Comment:
What is the behavior is format is undefined?

Reply:
Does TCGC uses a union, and do we currently handled all cases? If yes, we may remove this `default` case.

`default` was there for http lib, to my memory the reason is theirs is a string.

Reply:
```
export type CollectionFormat = "multi" | "csv" | "ssv" | "tsv" | "pipes" | "simple" | "form";

export interface SdkQueryParameter extends SdkModelPropertyTypeBase {
    kind: "query";
    collectionFormat?: CollectionFormat;
    serializedName: string;
    correspondingMethodParams: SdkModelPropertyType[];
    explode: boolean;
}
```

TCGC definition is above, we should have handled all the cases, so I can remove default.